src/hardware/gus.cpp

```diff
@@ -937,34 +938,26 @@ void Gus::SetupEnvironment(uint16_t port, const char* ultradir_env_val)
 	assert(dma1 < 10 && dma2 < 10);
 	assert(irq1 <= 12 && irq2 <= 12);
 
-	const std::string at_set = "@SET";
+	// Prepare AUTOEXEC.BAT lines
+	if (!autoexec_lines.at(0)) {
+		autoexec_lines.at(0) = std::make_unique<AutoExecObject>();
+	}
+	if (!autoexec_lines.at(1)) {
+		autoexec_lines.at(1) = std::make_unique<AutoExecObject>();
+	}
 
 	// ULTRASND variable
 	char ultrasnd_env_val[] = "HHH,D,D,II,II";
 	safe_sprintf(ultrasnd_env_val, "%x,%u,%u,%u,%u", port, dma1, dma2, irq1, irq2);
-	const auto ultrasnd_line = at_set + " " + ultrasnd_env_name + "=" +
-	                           ultrasnd_env_val;
-	autoexec_lines.at(0) = std::make_unique<AutoexecObject>(ultrasnd_line);
+	autoexec_lines[0]->SetVariable(ultrasnd_env_name, ultrasnd_env_val);
 
 	// ULTRADIR variable
-	const auto ultradir_line = at_set + " " + ultradir_env_name + "=" +
-	                           ultradir_env_val;
-	autoexec_lines.at(1) = std::make_unique<AutoexecObject>(ultradir_line);
-
-	if (first_shell) {
-		first_shell->SetEnv(ultrasnd_env_name, ultrasnd_env_val);
-		first_shell->SetEnv(ultradir_env_name, ultradir_env_val);
-	}
+	autoexec_lines[1]->SetVariable(ultradir_env_name, ultradir_env_val);
```

Comment:
It's an existing pattern, but I find this indexed line access business weird. Probably I could figure it out if I spent like a good 5-10 minutes on it, but at a glance I'm confused:

- Why the indexed access at line 0 and 1? Aren't we just appending lines to the config here? If so, why this weird API? I'd want expect to see some `autoexec.appendLine(...)` calls. So if we're just appending, this API is not great.
- What if there's something at lines 0 and 1 already? Will we overwrite the contents of those lines?

So given you're overhauling the whole autoexec handling apparently, could you introduce a better API than this low-level line mangling?

Reply:
I have simplified the variable API, now it is just a single call - no objects needed.